First directory restructuring

trxe opened the pull request:
Always check if there is an existing folder with the type of asset you are adding before you create your new folder with assets. All 3rd-Party assets should be under the correct subcategory in the `3rd-Party/` folder. 

Also note Towers and Bullets are in separate folders under `Towers/`

Some clarifications:
`Materials/`: .mat only
`Models/` : all model files (.fbx, .blend etc.)
`Prefabs/`: .prefab only
`Scripts/`: .cs only
`Textures/` : all images (.png, .jpg etc.)

trxe committed:
First directory restructuring

trxe committed:
move bullet prefabs out

trxe committed:
renaming materials and moving textures

trxe committed:
make TowerInfo Scriptable Object

tanwayne890 commented:
i played the `Menu` scene, works fine. all my files are in correct directories. Thanks @trxe 

Good to note that EZ Camera Shake is on path `Assets/3rd-Party/Scripts/EZ Camera Shake`

trxe committed:
create ScriptableObjects for cannon

trxe committed:
modify Shop related methods

trxe committed:
set Waypoint prefab in MapGenerator

trxe committed:
move ScriptableObjects to folder

trxe committed:
Merge branch 'directory-structure' into refactoring/towers

trxe committed:
re-enable tower building in BuildManager

trxe committed:
rename Player to InkManager, begin refactor of Build system

trxe committed:
temporarily disable tooltips, handle upgrades and elements

trxe committed:
Creation of Cannons

trxe committed:
create tower scripts, ballista prefabs

trxe committed:
Add Ballista ScriptableObjects

trxe committed:
add ink reduction to upgrades/element swap

trxe committed:
Set upgrade numbers, complete Spikey implementation

trxe committed:
complete migration of ShopItemUI

tanwayne890 committed:
Refactor waves (#200)

* change rounds to waveCounter

* rename GameManager to GameStateManager

* waveSpawner to include waveCounter, remove from GameStateManager

* remove comments + run formatter (reset brackets position)

* save shader scene (to assign wave counter text to waveSpawner)

* save shader scene 16-10

* revert m_indirectSpecularColor

* change comment: on what is triggering OnEnable

* refactor PauseGame

* remove "reset timer"

* "pause if player is viewing tutorial"

* remove "disable this script"

* remove "keep track of waves survive"

* remove todo

* remove update(), update and start comments

* remove update comment

* rename GameOverManager to GameOver

* add comments on retry

* remove update comments

* rename Gameover into GameOverManager

* "Enabled when game is running"

trxe committed:
nodeUI reduce repeating code and fix ice-water order bugs

trxe committed:
Add UI Camera

trxe committed:
fix stats and tooltips

tanwayne890 committed:
format wave

tanwayne890 committed:
rename wave to waveManager

trxe committed:
Merge pull request #203 from CS3247-Game-Development-Team-6/refactoring/nodeui-tooltip

Enabling tooltips on NodeUI

trxe committed:
Merge pull request #202 from CS3247-Game-Development-Team-6/refactoring/tower-prefabs

Tower building

trxe committed:
Merge pull request #201 from CS3247-Game-Development-Team-6/refactoring/towers

Setup new Tower system

trxe committed:
Merge pull request #204 from CS3247-Game-Development-Team-6/rename-wave

Rename wave

tanwayne890 committed:
fix water element (#205)

tanwayne890 committed:
Refactor `effect status` for enemy (#206)

* code style: enemy.cs

* refactor chill and format effectManager

* fix water element

* enum every effect status

* enemy rearrange seq and remove unnecesary comments

* remove unecessary comments for effectManager

* all CAPS

* move effectStatus outside of enemy

* SetEnemyVisibility

* private endPath

* private getnextwaypoint

* private die

* private start and update

trxe committed:
Fix tower-refactor scene

trxe committed:
Merge branch 'directory-structure' of https://github.com/CS3247-Game-Development-Team-6/Doodles into directory-structure

tanwayne890 committed:
Use scriptable objects for enemy (#207)

* code style: enemy.cs

* refactor chill and format effectManager

* fix water element

* enum every effect status

* enemy rearrange seq and remove unnecesary comments

* remove unecessary comments for effectManager

* change private fields

* add enemyinfo script and enemyfast scriptableobjects

* remove init health etc and use scriptable objects field

* save enemyinfos

* code style enemyshooting

* code style enemybullet

* rename "..."enemy and add enemyShootingInfo

* enemy range and firerate

* bullet speed and damage

* remove enemybullets for fast, simple and tough. add enableCameraShake

* private update target

Co-authored-by: Wong Pei Xian <peixianwong@gmail.com>

Co-authored-by: Wong Pei Xian <peixianwong@gmail.com>

trxe committed:
make curvePrefab and straightPrefab into nodes

trxe committed:
conditional build

trxe committed:
greyed out invalid tiles for placement on hover

Fenway17 committed:
Added player dash
- added new scene to test player dash
- changed player movement to attempt to fix clipping

Bugs:
- sometimes still clips terrain at certain angles

tanwayne890 committed:
Add spiders (#208)

* add elementEffectType

* Add elementType into enemyInfo (scriptable object), recalculate damage

* use example enemy: elementalMelee

* add all spiders and debug log

* add damagetext when enemy receive damage

* Add `SpiderScene-prototype` (#211)

* add cemetery halloween set

* add prototype scene with spider bg, base and necessary albedo

* remove cemetery halloween set (due to high poly and colourful)

* add deadtrees, bones, skull, eggs

* add spider animator, spider models and prefabs

* Spider model fix

- Fix textures for spider models
- Fix Polygon face orientation for spider models

* gather all spider prefabs into one folder

* move materials and fbms into Assets/Materials/Enemy/Spiders

* add spider level info for testing, fix FireMeleeSpider name, adjust lighting

* remove bizulkaproduction sample scene

* remove explosiveLLC

Co-authored-by: Danh Nguyen <ducdanh24112001@gmail.com>
Co-authored-by: trxe <peixianwong@gmail.com>

* change damageText colour green

* add elementInfo into EnemyInfo, add weakness into elementInfo

* add damage multiplier into EnemyInfo

Co-authored-by: Danh Nguyen <ducdanh24112001@gmail.com>
Co-authored-by: trxe <peixianwong@gmail.com>

trxe committed:
Merge branch 'directory-structure' into map-refactor

trxe committed:
add default allowedCellTypes in CanBuildTower

trxe committed:
Merge pull request #214 from CS3247-Game-Development-Team-6/map-refactor

Allow building of Towers on Path

trxe committed:
basic chunks implementation

trxe committed:
debugged scoring function

tanwayne890 committed:
add tower slot and inventory with workable select and deselect. a letsbuild button enabled on at atleast 1 tower

trxe committed:
fix path generation bugs and orientation of curve/straight prefabs

Timothyoung97 committed:
Add 3rd party asset for particle effect

Credit:
https://assetstore.unity.com/packages/vfx/particles/simple-fx-cartoon-particles-67834

Timothyoung97 committed:
Add basic damage behaviour to tower

Fenway17 committed:
Modified fog to have an updated hitbox to accomodate dashing

Fenway17 committed:
Added skeleton for gel explosion tower

Timothyoung97 committed:
Modify FX Smoke

Make FX Smoke to look more like fire then turn into smoke

Timothyoung97 committed:
Add new Tower Dev Scene

Timothyoung97 committed:
Relocate FX_Smoke from pre built gameobject to store in towerinfo

Timothyoung97 committed:
Add FX_Smoke to all tower scriptable objects

Timothyoung97 committed:
Restore CannonTower

Timothyoung97 committed:
Add update callback to Tower.cs

Fenway17 committed:
Added subclass for bullet for gel explosion tower

- currently the slow will conflict with freezing, require changes to enemy script

Timothyoung97 committed:
Add TowerInfo for 3 towers

Add health and decayRate

Timothyoung97 committed:
Migrate base damage behaviour to tower class

Timothyoung97 committed:
Restore 0_CannonBase.prefab

Fenway17 committed:
Fixed bugs in code

Timothyoung97 committed:
Add healthBarPrefabs for all tower's scriptable object

Timothyoung97 committed:
Add health bar instantiation in Tower.cs

Timothyoung97 committed:
Add 3rd party sound effects

Timothyoung97 committed:
Create and assign damage sound effect prefab to Tower's scriptable objects

Timothyoung97 committed:
Add damage sound effect to towers

Timothyoung97 committed:
Assign Damage audio prefab to rest of the towers

Timothyoung97 committed:
Add tower fix UI button

tanwayne890 committed:
canvas in-game

Eriksen2411 committed:
Spider Animation Integration

- Integrate  created Blender models, rigs and animations with ingame object and animator controller

Eriksen2411 committed:
Spider animation integration

Fenway17 committed:
Implemented extra effect manager to handle gel slow and future effects not tied to elements

- updated enemy script to reflect this
- updated gel tower and bullet scripts to reflect this
- added extra effect manager script to ALL enemy prefabs

trxe committed:
fix barriers, add waypoints

Timothyoung97 committed:
Add fix tower image

trxe committed:
wavespawning, bug when a wave is cleared

Timothyoung97 committed:
Add skeleton tower fix code to NodeUI

Timothyoung97 committed:
Update TowerDamageDevScene.unity

For testing purpose

Timothyoung97 committed:
Add damageFixCost to each tower

Timothyoung97 committed:
Add functionality for tower fix

Timothyoung97 committed:
Change behaviour for tower fix

Such that player can fix the tower even if not dead

Timothyoung97 committed:
Fix VFX & Audio delay issue

Timothyoung97 committed:
Restore 0_CannonBase to default from testing

Timothyoung97 committed:
Test and extend tower damage behaviour to other towers

Fenway17 committed:
Created gel tower assets using dummy cannon assets

tanwayne890 committed:
load spider scene

tanwayne890 committed:
add choose your tower

trxe committed:
EnemyEffectManager

Fenway17 committed:
Merge pull request #223 from CS3247-Game-Development-Team-6/add-gel-tower-effect-manager

Suggested changes to ExtraEffectManager --> EnemyEffectManager

tanwayne890 committed:
cleanup

Fenway17 committed:
Reimplemented enemy active effect system

- Reimplemented gel effect
- Changed enemy prefab to reflect new manager for the effects

Fenway17 committed:
shifted variables around

trxe committed:
add files for separation

Timothyoung97 committed:
Remove unused imports

tanwayne890 committed:
add shopTowerUI to prefab

trxe committed:
creates a new multiplier for the enemy speed to fix bug

trxe committed:
rename EnemyActiveEffectsManager to EnemyActiveEffects

tanwayne890 committed:
use correct ink values

tanwayne890 committed:
able to select and build tower

Timothyoung97 committed:
Add function to check if tower is in full health

Timothyoung97 committed:
Add boolean checks for Node UI to reflect fix tower cost

Timothyoung97 committed:
Add dynamic change to fixDamageTowerCost in UI and ink required

Timothyoung97 committed:
Refactor smokePrefab to set parent to current tower

trxe committed:
handle basic progression of chunks

trxe committed:
add all dialogues from dialogue-implementation

tanwayne890 committed:
start button to loadout scene (mainmenu)

tanwayne890 committed:
move ShopTowerUI to cardManager

tanwayne890 committed:
destroy towerSelectionManager after loadout scene

trxe committed:
add towers, haven't added bullet prefabs per tower

tanwayne890 committed:
getDDOLObjects,loadout bg

Timothyoung97 committed:
Set all damageFixFactor to 1 as default

tanwayne890 committed:
Merge branch 'dialogue-scenes-px' into feature-loadout

# Conflicts:
#	UserSettings/EditorUserSettings.asset

Timothyoung97 committed:
Change variable naming for fixButton

Fenway17 committed:
Merge pull request #224 from CS3247-Game-Development-Team-6/add-gel-tower-fix

[Gel Tower] Fix speed bug and cleanup+generalize

Timothyoung97 committed:
Rename smoke prefab

Timothyoung97 committed:
Change smoke, healthBar, sound prefab assignment

Each tower to refer to prefabs stored in TowerManager.instance

tanwayne890 committed:
install yarn-spinner

Timothyoung97 committed:
Update NodeUI.prefab

Fenway17 committed:
Fixed asset for bullet prefab

Fenway17 committed:
Fixed bullet prefab assignment

Timothyoung97 committed:
Bug fix: Spikey Tower's bullets not shown

Error: respective scriptable object's speed not set

trxe committed:
started to migrate OnScreenTutorialUI

trxe committed:
create alpha level

trxe committed:
Merge branch 'map-chunks-spawn-handling' into map-chunk-create-level

Timothyoung97 committed:
Merge branch 'directory-structure' into ConResolve-Dev-Tower-DamageMechanics

Fenway17 committed:
Added stats to gel explosion tower scriptable objects

trxe committed:
Tower descriptions page

trxe committed:
Merge branch 'add-gel-tower' into map-chunk-create-level

trxe committed:
Merge remote-tracking branch 'origin/Debug-SpikeyTower-BulletNotShown' into map-chunk-create-level

tanwayne890 committed:
menu -> tutorial dialog -> spider dialog -> loadout -> spiderlevel

trxe committed:
GelPrefab ok

tanwayne890 committed:
add audio bird chirping

trxe committed:
Merge branch 'add-gel-tower' into map-chunk-create-level

tanwayne890 committed:
Merge branch 'player-dash-1' into integrating-dialogs

# Conflicts:
#	UserSettings/EditorUserSettings.asset

trxe committed:
Merge branch 'ConflictResolve-Tower-DamageMechanics' into map-chunks-level-creation

trxe committed:
shift alpha scene location

trxe committed:
shifted SpiderScene-alpha

tanwayne890 committed:
Merge branch 'Dev-Tower-DamageMechanics' into integrating-dialogs

# Conflicts:
#	UserSettings/EditorUserSettings.asset

tanwayne890 committed:
Merge branch 'map-chunks-level-creation' into integrating-dialogs

# Conflicts:
#	ProjectSettings/EditorBuildSettings.asset
#	UserSettings/EditorUserSettings.asset

trxe committed:
set MapInfo in inkManager

tanwayne890 committed:
build settings for dialogues

trxe committed:
Merge branch 'integrating-dialogs' into map-chunks-tutorial

trxe committed:
add visible barriers and set OnScreenTutorialUI on each wave clear

trxe committed:
OnScreenTutorial flag

trxe committed:
debugged the instructions and added demo to DebugLevel

trxe committed:
setup for deployment

trxe committed:
default tower as first in shop, adjust post-process outline and loading screen

trxe committed:
remove buggy loading

trxe committed:
Merge branch 'integrating-dialogs' into build-settings

tanwayne890 committed:
use spiderThemeEmptyTile

tanwayne890 committed:
remove retry and menu button in winUI and pauseCanvas

tanwayne890 committed:
loadingBar

Fenway17 committed:
Created skeleton for first tutorial level and tutorial map chunk

Fenway17 committed:
Finished up tutorial level

- Added part of spider level

Fenway17 committed:
Rebalnced waves for spider level

Fenway17 committed:
Expanded on spider level chunk 2 waves

Fenway17 committed:
Added third chunk for spider level
- TODO: finish up third chunk

Fenway17 committed:
Added notes to spider level chunk 3

Fenway17 committed:
Refined tutorial notes to be more concise and clear

tanwayne890 committed:
fix enemy attacking base before actually reaching

tanwayne890 committed:
Merge branch 'fix-enemy-range' into add-tutorial-and-spider-level

tanwayne890 committed:
refined bass,fast,simple,tough enemies stats

tanwayne890 committed:
reduce tutorial chunk  enemy

tanwayne890 committed:
Merge branch 'build-settings' into add-tutorial-and-spider-level

tanwayne890 committed:
increase ink gained for spiders

tanwayne890 committed:
menu -> tutorial  dialog -> tutorial level -> spider dialog -> spider level

tanwayne890 committed:
remove debug log for gel explosion

tanwayne890 committed:
fix enemy boss hp canvas

tanwayne890 committed:
fix camera movement

trxe committed:
fix restoring tower and gel tower health, increase barrier movement speed

trxe committed:
OnScreenTutorialUI fix progression

trxe committed:
fix all OnScreenTutorial bugs

trxe committed:
fix invisible text bug

trxe committed:
Merge pull request #234 from CS3247-Game-Development-Team-6/add-tutorial-and-spider-level

Alpha-ready build ver 1
